server/game/cards/14-FotS/ACaskOfAle.js

```diff
@@ -5,56 +5,71 @@ class ACaskOfAle extends DrawCard {
     setupCardAbilities() {
         this.reaction({
             when: {
-                afterChallenge: event => event.challenge.isMatch({ winner: this.controller, challengeType: 'power' })
+                afterChallenge: event => (event.challenge.isMatch({ winner: this.controller, challengeType: 'power' }) 
+                    && this.game.allCards.some(card => (card.location === 'active plot', 'faction', 'play area' && card.getPower()>0)))
             },
-            targets: {
-                from: {
-                    activePromptTitle: 'Select card with power',
-                    cardCondition: card => ['active plot', 'faction', 'play area'].includes(card.location) && card.power > 0,
-                    cardType: ['attachment', 'character', 'faction', 'location', 'plot']
-                },
-                to: {
-                    activePromptTitle: 'Select card to place power',
-                    cardCondition: (card, context) => (
-                        ['active plot', 'faction', 'play area'].includes(card.location) &&
-                        (!context.targets.from || card !== context.targets.from && card.controller === context.targets.from.controller)
-                    ),
-                    cardType: ['attachment', 'character', 'faction', 'location', 'plot']
-                }
+            target: {
+                mode: 'exactly',
+                numCards: 2,
+                activePromptTitle: 'Select 2 cards',
+                singleController: true,
+                cardCondition: card => ['active plot', 'faction', 'play area'].includes(card.location),
+                cardType: ['attachment', 'character', 'faction', 'location', 'plot'],
+                gameAction: 'movePower'
```

Comment:
This doesn't guarantee that cards chosen are from a player that has power. Is that necessary, or just that power is somewhere on the board?

Reply:
I asked Miha (Odrl) about it and you should be able to play it as long as there is power in play (I guess you could play it for some trigger like Tom or DWDW and then let the effect fizzle). The problem before was that it would allow you to select the cards from the opponent and yours, because the first card was read as undefined when selecting the second one. I think there is a similar problem with Richard Horpe that lets you move power back to the same card.